Packs/qualys/Integrations/Qualysv2/Qualysv2.py

```diff
@@ -2902,6 +2902,41 @@ def get_detections_from_hosts(hosts):
     return fetched_assets, False
 
 
+def send_assets_and_vulnerabilities_to_xsiam(
+    assets: list,
+    vulnerabilities: list,
+    cumulative_assets_count: int,
+    cumulative_vulns_count: int,
+    has_next_page: bool,
+    snapshot_id: str,
+) -> None:
+    """Sends assets and vulnerabilities to their respective snapshot datasets.
+
+    Args:
+        assets (list): List of host detections (assets) dictionaries.
+        vulnerabilities (list): List of vulnerabilities dictionaries.
+        cumulative_assets_count (int): Total count of assets collected since resetting last run object.
+        cumulative_vulns_count (int): Total count of vulnerabilities collected since resetting last run object.
+        has_assets_next_page (bool): Whether there is a next assets page url (indicates more results).
+        snapshot_id (str): Snapshot ID of the dataset (use the same snapshot ID to add more data to the same dataset snapshot).
+    """
+    demisto.debug(f'Sending {len(assets)} assets to XSIAM. '
+                  f'Total assets collected so far: {cumulative_assets_count}')
+
+    total_assets_to_report = 1 if has_next_page else cumulative_assets_count   # set 1 if not done pulling
```

Comment:
Why are we sending just 1 if there is next page? why not send everything we got so far?
Lets document this decision.

Reply:
Added a clearer comment to explain the rational behind `total_assets_to_report` and `total_vulns_to_report` and documented internally.